merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/engine/TaskId.java

```diff
@@ -3,7 +3,7 @@
 import java.util.UUID;
 
 /** A typed wrapper for task IDs. */
-/*package-local*/ record TaskId(String id) {
+/*package-local*/ public record TaskId(String id) {
   public static TaskId generate() {
     return new TaskId(UUID.randomUUID().toString());
   }
```

Comment:
Just so you and @mattdailis are aware, there is going to be a collision with this class.  @mattdailis changed the structure to this:

```
/** A typed wrapper for task IDs. */
public record TaskId(String id) implements Scheduler.TaskId {
  public static TaskId generate() {
    return new TaskId(UUID.randomUUID().toString());
  }
}
```

Reply:
Hmm... I'm trying to remember why I made it public. It might not need to be for computed attributes.

Also if we're getting rid of `waitFor` as suggested by Jonathan, it won't need to implement Scheduler.TaskId either 👍

Reply:
Peripheral question about this: why does `Scheduler.spawn` return a `String` version of the `TaskId` but not the object used in the engine? I had to add a nasty search in the `SimulationEngine` to be able to get the `TaskId` object corresponding to the same id.

Reply:
`Scheduler.spawn` is the interface used by the model - we do not want the model to be able to use task ids for anything other than calling `waitFor(taskid)`.

Since TaskId is a record, you actually don't really need to find the original object - you can make `new TaskId(String)` and it will behave in all the same ways (same hashcode, same equals). Just don't use `==` on it.

We are getting rid of spawn returning a String - it's going to return `void` quite soon. The driver will need to get its task ids some other way 🤔

Reply:
There's also the package dependency reason: `TaskId` is in `merlin-driver` (where the engine lives). The model does not have access to the engine's types. The only types that cross the boundary between the model and the engine live in `merlin-sdk`, in the `protocol` package.

Reply:
That will probably change how **everything** is done in this incremental simulation, from ControlTask to getting durations back. We might want to wait for that change to be merged.

Reply:
@mattdailis my use case was getting execution states from the SimulationEngine.tasks map. And creating new TaskId for that did not work because they are different objects.

Reply:
Something else might have been confounding things, because every `record` automatically generates an overridden `equals()` and `hashCode()` that invoke the same methods over each of their components. The `Map` class uses these methods, not reference equality, on their keys.

Reply:
We frequently construct new `TaskId` wrappers around the same `String` ID in the engine itself -- see [this line](https://github.com/NASA-AMMOS/aerie/blob/develop/merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/engine/SimulationEngine.java#L249). If it didn't work this way, tasks waiting on their children would never awaken.